File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;

namespace WindowsFormsApplication13
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            
            if (xpCollection1.Count == 0) {
                Random r = new Random();
                for (int i = 0; i < 10; i++) {
                    MyObject b = new MyObject();
                    b.BLOCKCODE = r.Next(10000).ToString();
                    if (i % 2 != 0)
                        b.FORMERLYWAS = xpCollection1[r.Next(xpCollection1.Count)] as MyObject;
                    b.Save();
                    xpCollection1.Add(b);
                }
            }
            dataGridView1.DataSource = xpCollection1;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            XPCollection<MyObject> c = new XPCollection<MyObject>(CriteriaOperator.Parse("not RelatedObjects[].Exists()"), null);
            dataGridView2.DataSource = c;
        }
    }

    public class MyObject : XPBaseObject { 
        //PSCID,PSCNAME,BLOCKCODE,FORMERLYWAS from PSCBLOCK
        [Key(true)]
        public int PSCID;
        
        [Association("ID_FORMERLYWAS")]
        public MyObject FORMERLYWAS;

        [Association("ID_FORMERLYWAS", typeof(MyObject))]
        public XPCollection<MyObject> RelatedObjects {
            get { return GetCollection<MyObject>("RelatedObjects"); }
        }
        public string PSCNAME;
        public string BLOCKCODE;

    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Namespace WindowsFormsApplication13
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

			If xpCollection1.Count = 0 Then
				Dim r As New Random()
				For i As Integer = 0 To 9
					Dim b As New MyObject()
					b.BLOCKCODE = r.Next(10000).ToString()
					If i Mod 2 <> 0 Then
						b.FORMERLYWAS = TryCast(xpCollection1(r.Next(xpCollection1.Count)), MyObject)
					End If
					b.Save()
					xpCollection1.Add(b)
				Next i
			End If
			dataGridView1.DataSource = xpCollection1
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim c As XPCollection(Of MyObject) = New XPCollection(Of MyObject)(CriteriaOperator.Parse("not RelatedObjects[].Exists()"), Nothing)
			dataGridView2.DataSource = c
		End Sub
	End Class

	Public Class MyObject
		Inherits XPBaseObject
		'PSCID,PSCNAME,BLOCKCODE,FORMERLYWAS from PSCBLOCK
		<Key(True)> _
		Public PSCID As Integer

		<Association("ID_FORMERLYWAS")> _
		Public FORMERLYWAS As MyObject

		<Association("ID_FORMERLYWAS", GetType(MyObject))> _
		Public ReadOnly Property RelatedObjects() As XPCollection(Of MyObject)
			Get
				Return GetCollection(Of MyObject)("RelatedObjects")
			End Get
		End Property
		Public PSCNAME As String
		Public BLOCKCODE As String

	End Class
End Namespace
